File: src/packets/PacketS2CZoneParameter.js
const PacketS2C = require("./PacketS2C");
const parameterIsSigned = require("./parameterIsSigned")

/**
 * Server -> Client
 * ID = 0x0B
 * Zone Parameter
 * Sends extra parameter values
 * Data:
 *      (Unsigned Char) Controller ID
 *      (Unisgned Char) Zone ID
 *      (Unsigned Char) Paramter ID
 *      (Unsigned/Signed Char) Paramter Value
 */

class PacketS2CZoneParameter extends PacketS2C {
    constructor(ctrlrID, zoneID, paramterID, value) {
        super();

        this._buffer.writeUInt8(ctrlrID);
        this._buffer.writeUInt8(zoneID);
        this._buffer.writeUInt8(paramterID);

        if (parameterIsSigned(paramterID)) {
            this._buffer.writeInt8(value);
        }
        else {
            this._buffer.writeUInt8(value);
        }
    }

    getID() {
        return 0x0B;
    }
}

module.exports = PacketS2CZoneParameter;

File: src/RNetPi.Core/Packets/PacketS2CZoneParameter.cs
using RNetPi.Core.Utilities;

namespace RNetPi.Core.Packets;

/// <summary>
/// Server -> Client
/// ID = 0x0B
/// Zone Parameter
/// Sends extra parameter values
/// Data:
///     (Unsigned Char) Controller ID
///     (Unsigned Char) Zone ID
///     (Unsigned Char) Parameter ID
///     (Unsigned/Signed Char) Parameter Value
/// </summary>
public class PacketS2CZoneParameter : PacketS2C
{
    public const byte ID = 0x0B;

    public PacketS2CZoneParameter(byte controllerID, byte zoneID, byte parameterID, object value)
    {
        Writer.Write(controllerID);
        Writer.Write(zoneID);
        Writer.Write(parameterID);

        if (ParameterUtils.IsParameterSigned(parameterID))
        {
            Writer.Write((sbyte)Convert.ToSByte(value));
        }
        else if (ParameterUtils.IsParameterBoolean(parameterID))
        {
            Writer.Write((byte)((value is bool boolValue && boolValue) ? 1 : 0));
        }
        else
        {
            Writer.Write(Convert.ToByte(value));
        }
    }

    public override byte GetID() => ID;
}